File: backend/src/Blindify.Domain/Enums/RoundCible.cs
namespace Blindify.Domain.Enums;

/// <summary>Ce qui est demandé au joueur pour un round donné — tiré aléatoirement au démarrage
/// (voir RoundService.DemarrerRound), affiché aux joueurs pour lever l'ambiguïté quand un morceau
/// a plusieurs auteurs.</summary>
public enum RoundCible
{
    Titre,
    Auteur,

    /// <summary>Forcée pour les morceaux taggés "disney" (voir RoundService.DemarrerRound) : le
    /// titre réel de la chanson ou l'artiste crédité (souvent la voix/l'acteur) sont imprévisibles
    /// à deviner — le film dont est tiré le morceau (Track.Album nettoyé, voir FilmNameResolver)
    /// est la question naturelle pour ce type de contenu.</summary>
    Film,

    /// <summary>V2, section 12.5 — "En quelle année est sorti ce morceau ?". Éligible seulement si
    /// Track.Year est connu ; jamais tirée pour les morceaux "disney" (toujours Film). Voir
    /// GameConfig.PoidsCibleAnnee et ScoringService (scoring à la proximité en mode saisie).</summary>
    Annee
}

File: app/lib/models/round_cible.dart
/// Miroir de `Blindify.Domain.Enums.RoundCible` — ce qui est demandé au joueur pour
/// ce round (tiré aléatoirement côté serveur, voir RoundService.DemarrerRound).
/// `film` est forcée pour les morceaux "disney" (titre réel/auteur crédité imprévisibles).
/// `annee` (V2, section 12.5) : éligible seulement si Track.Year est connu côté serveur.
enum RoundCible { titre, auteur, film, annee }

extension RoundCibleJson on RoundCible {
  static RoundCible fromJson(String value) {
    switch (value) {
      case 'Titre':
        return RoundCible.titre;
      case 'Auteur':
        return RoundCible.auteur;
      case 'Film':
        return RoundCible.film;
      case 'Annee':
        return RoundCible.annee;
    }
    throw ArgumentError('RoundCible inconnue reçue du serveur : $value');
  }

  String get label {
    switch (this) {
      case RoundCible.titre:
        return 'le titre';
      case RoundCible.auteur:
        return "l'artiste";
      case RoundCible.film:
        return 'le film';
      case RoundCible.annee:
        return "l'année";
    }
  }
}
